vulkaninfo/outputprinter.h

```diff
@@ -51,20 +51,28 @@ std::string to_string(const std::array<uint8_t, 8> &uid) {
     return ss.str();
 }
 
-std::string VkVersionString(uint32_t version) {
-    uint32_t major = version >> 22;
-    uint32_t minor = (version >> 12) & 0x3ff;
-    uint32_t patch = version & 0xfff;
-    return std::to_string(major) + "." + std::to_string(minor) + "." + std::to_string(patch);
+std::ostream &operator<<(std::ostream &out, const VkConformanceVersion &v) {
+    return out << static_cast<unsigned>(v.major) << "." << static_cast<unsigned>(v.minor) << "."
+               << static_cast<unsigned>(v.subminor) << "." << static_cast<unsigned>(v.patch);
 }
 
-std::string VkVersionString(VulkanVersion v) {
-    return std::to_string(v.major) + "." + std::to_string(v.minor) + "." + std::to_string(v.patch);
-}
-
-std::string VkConformanceVersionString(const VkConformanceVersion &c) {
-    return std::to_string(c.major) + "." + std::to_string(c.minor) + "." + std::to_string(c.subminor) + "." +
-           std::to_string(c.patch);
+// Vendor specific driverVersion mapping scheme
```

Comment:
I'm iffy on this. It is arguably more correct key off `VkPhysicalDeviceDriverProperties::driverID` than `VkPhysicalDeviceProperties::vendorID`, but that struct already has a string field with the human-readable driver version. I'm assuming mesa NVK driver will report the NVIDIA vendor ID and get a mangled version as a result.

Reply:
Thats an excellent point. I think using VkPhysicalDeviceDriverProperties::driverID is superior just because it allows more specificity. I would likely need a fallback path for 1.0 drivers though, which means this code isn't completely useless.

As far as the NVK driver goes, I think using driverID will hopefully circumvent this issue. I assume NVK will use the Mesa versioning scheme, not the NVidia one, so using the Vulkan API version would be correct.

FYI the driverVersion formatting was taken from https://github.com/SaschaWillems/vulkan.gpuinfo.org/blob/cc83072c855ea00ad4c5f85148e822c094668fcf/includes/functions.php#L364

Reply:
That's an interesting snippet from gpuinfo.org and it probably predates VK_KHR_driver_properties. I would still be inclined to scrap the uint32_t -> semantic version string conversion because 1) There isn't enough information to do it accurately without VkPhysicalDeviceDriverProperties(KHR) and 2) Given VkPhysicalDeviceDriverProperties, the driver has already done a string conversion for you. I guess that means I'm ok with old drivers have semi-opaque versions, that was probably what motivated the introduction of VK_KHR_driver_properties in the first place.

Reply:
I think this refers to changing what was historically an integer field to a string field... I'm against this in principle, as it's a unnecessary compatibility break.  Adding a new field with a formatted/mangled/whatever value is more acceptable than breaking downstream users who are trying to use `vulkaninfo -j` for production (like, our CI, which will certainly break).

Reply:
I just double checked the schema, and changing the apiVersion & driverVersion in the profiles output fails the schema, so I'll definitely be changing those fields back to using the uint32_t's instead of the human readable strings.

Reply:
@mikes-lunarg Something to note about `VkPhysicalDeviceDriverProperties::driverInfo` is that it will likely contain a version, but can contain a lot more information in it, including characters which break the json output in the first place (hence the main purpose of this PR). Saying 'the driver has already done a string conversion' isn't necessarily true, since it could *not* print the version in the `driverInfo`.

That said, `driverID` is significantly more robust than `vendorID`, so definitely should be preferred when available.

Reply:
Ok I can get on board with this. The new implementation opts new drivers out of the legacy vendor matching if they provide `driverID`. A nice side-effect is a more accurate profile name on my NVIDIA device!

Reply:
Yep, much more 'robust'. I also made sure to only apply the 'formatted' driverVersion to the text/html/vkconfig output, as the profile json schema requires the field be an int, not a string.

I kinda feel bad now for thinking all driverVersion fields used the Vulkan API version format, since that is patently not the case!